### ISSUE TITLE
我通过猎物踪迹寻找猎物时崩溃

### ISSUE THREAD

**Issue by yue-git** [penguin0616/insight#17]

**Comment by yue-git** [comment]
我通过猎物踪迹寻找猎物时崩溃
crashed when i looking for prey through the prey trail
```
[00:11:34]: ...s/workshop-2081254154/scripts/descriptors/hunter.lua:23: attempt to index global 'TheWorld' (a nil value)
LUA ERROR stack traceback:
E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:23 in (upvalue) GetAlternateBeastChance (Lua) <22-26>
E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:49 in (upvalue) GetHuntDataFromTrack (Lua) <38-51>
   inst = 119339 - dirtpile (valid:true)
   hunt = table: 1295DC90
E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:67 in (field) Describe (Lua) <53-87>
   inst = 119339 - dirtpile (valid:true)
   context = table: 3789D848
E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/modmain.lua:813 in (upvalue) GetEntityInformation (Lua) <778-913>
   entity = 119339 - dirtpile (valid:true)
   player = 100021 - wilson (valid:true)
   params = table: 3789DCA8
   assembled = table: 3789DA00
   player_context = table: 3789D848
   chunks = table: 3789D870
   prefab_descriptor = table: 128DFD98
E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/modmain.lua:952 in (upvalue) RequestEntityInformation (Lua) <916-1006>
   entity = 119339 - dirtpile (valid:true)
   player = 100021 - wilson (valid:true)
   params = table: 3789DCA8
   insight = table: 374F88D8
E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:713 in (method) SetString (Lua) <682-830>
   self =
      enabled = true
      focus_target = false
      InsightSetSize = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/widgets/text_lib.lua:23
      string = 斧头*
þ: 装备
 
 
 
      shown = false
      SetString = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:682
      callbacks = table: 4A070690
      parent = HoverText
      InsightGetSize = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/widgets/text_lib.lua:35
      focus_flow_args = table: 4A070D98
      inst = 109249 -  (valid:true)
      focus = false
      focus_flow = table: 4A070AC8
      name = Text
      children = table: 4A070758
      Hide = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:589
   text = 调查 可疑的土堆
   item = 119339 - dirtpile (valid:true)
E:/games/stream/steamapps/common/dont_starve/data/scripts/widgets/hoverer.lua:107 in (upvalue) oldOnUpdate (Lua) <23-129>
   self =
      callbacks = table: 4A070550
      followhandler = table: 4A070AF0
      inst = 109248 -  (valid:true)
      focus = false
      children = table: 4A0705F0
      focus_flow_args = table: 4A070730
      focus_target = false
      secondarytext = Text - þ: 烹饪
      text = Text - 斧头*
þ: 装备
 
 
 
      insightText = RichText - <null>
      enabled = true
      parent = Controls
      OnUpdate = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:674
      name = HoverText
      focus_flow = table: 4A070438
      shown = true
      owner = 100021 - wilson (valid:true)
      isFE = false
   using_mouse = true
   str = 调查 可疑的土堆
   colour = table: 14BB1FF8
   secondarystr = nil
E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:679 in (method) OnUpdate (Lua) <674-680>
   self =
      callbacks = table: 4A070550
      followhandler = table: 4A070AF0
      inst = 109248 -  (valid:true)
      focus = false
      children = table: 4A0705F0
      focus_flow_args = table: 4A070730
      focus_target = false
      secondarytext = Text - þ: 烹饪
      text = Text - 斧头*
þ: 装备
 
 
 
      insightText = RichText - <null>
      enabled = true
      parent = Controls
      OnUpdate = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:674
      name = HoverText
      focus_flow = table: 4A070438
      shown = true
      owner = 100021 - wilson (valid:true)
      isFE = false
   arg = nil
E:/games/stream/steamapps/common/dont_starve/data/scripts/frontend.lua:573 in (method) Update (Lua) <425-593>
   self =
      topblackoverlay = Image - images/global.xml:square.tex
      fade_title_out = false
      ignoreups = table: 124945F8
      overlayroot = overlayroot
      save_indicator_time_left = 0
      fade_title_time = 0
      page_repeat_time = 0.25
      OnMouseMove = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:350
      helptextbg = Image - images/global.xml:square.tex
      helptext = HelpText
      subtitle = Text - 
      updating_widgets_alt = table: 156FFC20
      title = Text - 
      scroll_repeat_time = -1
      topFadeHidden = false
      gameinterface = 100012 -  (valid:true)
      updating_widgets = table: 12495E08
      lastx = 641
      total_fade_time = 1
      lasty = 401
      autosave_enabled = true
      OnMouseButton = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:339
      delayovercb = false
      blackoverlay = Image - images/global.xml:square.tex
      screenstack = table: 12493B30
      num_pending_saves = 0
      fade_delay_time = false
      fade_time = 1.0038603658477
      saving_indicator = UIAnim
      save_indicator_fade_time = 0
      alpha = -2.3011524197258e-007
      tracking_mouse = true
      focus_locked = false
      fade_title_in = false
      screenroot = screenroot
      consoletext = Text - CONSOLE TEXT
      repeat_time = 0.049995800107718
      displayingerror = false
      helptexttext = Text - 
   dt = 0.016393899917603
   k = HoverText
   v = true
E:/games/stream/steamapps/common/dont_starve/data/scripts/update.lua:46 in () ? (Lua) <3-50>
   dt = 0.016393899917603

[00:11:34]: Last called Lua: luaWallUpdate
[00:11:34]: scripts/mainfunctions.lua(960,1) LUA ERROR stack traceback:
        E:/games/stream/steamapps/common/dont_starve/data/scripts/mainfunctions.lua(960,1)	
[00:11:34]: scripts/mainfunctions.lua(961,1) ...s/workshop-2081254154/scripts/descriptors/hunter.lua:23: attempt to index global 'TheWorld' (a nil value)
LUA ERROR stack traceback:
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:23 in (upvalue) GetAlternateBeastChance (Lua) <22-26>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:49 in (upvalue) GetHuntDataFromTrack (Lua) <38-51>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:67 in (field) Describe (Lua) <53-87>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/modmain.lua:813 in (upvalue) GetEntityInformation (Lua) <778-913>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/modmain.lua:952 in (upvalue) RequestEntityInformation (Lua) <916-1006>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:713 in (method) SetString (Lua) <682-830>
    E:/games/stream/steamapps/common/dont_starve/data/scripts/widgets/hoverer.lua:107 in (upvalue) oldOnUpdate (Lua) <23-129>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:679 in (method) OnUpdate (Lua) <674-680>
    E:/games/stream/steamapps/common/dont_starve/data/scripts/frontend.lua:573 in (method) Update (Lua) <425-593>
    E:/games/stream/steamapps/common/dont_starve/data/scripts/update.lua:46 in () ? (Lua) <3-50>
	
[00:11:34]: scripts/frontend.lua(773,1) SCRIPT ERROR! Showing error screen	
[00:11:34]: #...s/workshop-2081254154/scripts/descriptors/hunter.lua:23: attempt to index global 'TheWorld' (a nil value)
#LUA ERROR stack traceback:
@E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:23 in (upvalue) GetAlternateBeastChance (Lua) <22-26>
@E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:49 in (upvalue) GetHuntDataFromTrack (Lua) <38-51>
   inst = 119339 - dirtpile (valid:true)
   hunt = table: 1295DC90
@E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:67 in (field) Describe (Lua) <53-87>
   inst = 119339 - dirtpile (valid:true)
   context = table: 39A31260
@E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/modmain.lua:813 in (upvalue) GetEntityInformation (Lua) <778-913>
   entity = 119339 - dirtpile (valid:true)
   player = 100021 - wilson (valid:true)
   params = table: 39A31210
   assembled = table: 39A31378
   player_context = table: 39A31260
   chunks = table: 39A315F8
   prefab_descriptor = table: 128DFD98
@E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/modmain.lua:952 in (upvalue) RequestEntityInformation (Lua) <916-1006>
   entity = 119339 - dirtpile (valid:true)
   player = 100021 - wilson (valid:true)
   params = table: 39A31210
   insight = table: 374F88D8
@E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:713 in (method) SetString (Lua) <682-830>
   self =
      focus_flow_args = table: 4A070D98
      string = 斧头*
þ: 装备
 
 
 
      InsightSetSize = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/widgets/text_lib.lua:23
      focus_target = false
      callbacks = table: 4A070690
      enabled = true
      shown = false
      SetString = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:682
      InsightGetSize = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/widgets/text_lib.lua:35
      name = Text
      focus = false
      focus_flow = table: 4A070AC8
      inst = 109249 -  (valid:false)
      children = table: 4A070758
      Hide = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:589
   text = 调查 可疑的土堆
   item = 119339 - dirtpile (valid:true)
@E:/games/stream/steamapps/common/dont_starve/data/scripts/widgets/hoverer.lua:107 in (upvalue) oldOnUpdate (Lua) <23-129>
   self =
      callbacks = table: 4A070550
      inst = 109248 -  (valid:false)
      focus = false
      children = table: 4A0705F0
      focus_flow_args = table: 4A070730
      focus_target = false
      secondarytext = Text - þ: 烹饪
      text = Text - 斧头*
þ: 装备
 
 
 
      insightText = RichText - <null>
      enabled = true
      OnUpdate = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:674
      name = HoverText
      focus_flow = table: 4A070438
      shown = true
      owner = 100021 - wilson (valid:true)
      isFE = false
   using_mouse = true
   str = 调查 可疑的土堆
   colour = table: 14BB1FF8
   secondarystr = nil
@E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:679 in (method) OnUpdate (Lua) <674-680>
   self =
      callbacks = table: 4A070550
      inst = 109248 -  (valid:false)
      focus = false
      children = table: 4A0705F0
      focus_flow_args = table: 4A070730
      focus_target = false
      secondarytext = Text - þ: 烹饪
      text = Text - 斧头*
þ: 装备
 
 
 
      insightText = RichText - <null>
      enabled = true
      OnUpdate = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua:674
      name = HoverText
      focus_flow = table: 4A070438
      shown = true
      owner = 100021 - wilson (valid:true)
      isFE = false
   arg = nil
@E:/games/stream/steamapps/common/dont_starve/data/scripts/frontend.lua:573 in (method) Update (Lua) <425-593>
   self =
      topblackoverlay = Image - images/global.xml:square.tex
      fade_title_out = false
      ignoreups = table: 124945F8
      overlayroot = overlayroot
      save_indicator_time_left = 0
      fade_title_time = 0
      page_repeat_time = 0.25
      OnMouseMove = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:350
      helptextbg = Image - images/global.xml:square.tex
      helptext = HelpText
      subtitle = Text - 
      updating_widgets_alt = table: 156FFC20
      title = Text - 
      scroll_repeat_time = -1
      topFadeHidden = false
      gameinterface = 100012 -  (valid:true)
      updating_widgets = table: 12495E08
      lastx = 641
      total_fade_time = 1
      lasty = 401
      autosave_enabled = true
      OnMouseButton = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:339
      delayovercb = false
      blackoverlay = Image - images/global.xml:square.tex
      screenstack = table: 12493B30
      num_pending_saves = 0
      fade_delay_time = false
      fade_time = 1.0038603658477
      saving_indicator = UIAnim
      save_indicator_fade_time = 0
      alpha = -2.3011524197258e-007
      tracking_mouse = true
      focus_locked = false
      fade_title_in = false
      screenroot = screenroot
      consoletext = Text - CONSOLE TEXT
      repeat_time = 0.049995800107718
      displayingerror = false
      helptexttext = Text - 
   dt = 0
   k = HoverText
   v = true
@E:/games/stream/steamapps/common/dont_starve/data/scripts/frontend.lua:624 in (method) PushScreen (Lua) <604-629>
   self =
      topblackoverlay = Image - images/global.xml:square.tex
      fade_title_out = false
      ignoreups = table: 124945F8
      overlayroot = overlayroot
      save_indicator_time_left = 0
      fade_title_time = 0
      page_repeat_time = 0.25
      OnMouseMove = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:350
      helptextbg = Image - images/global.xml:square.tex
      helptext = HelpText
      subtitle = Text - 
      updating_widgets_alt = table: 156FFC20
      title = Text - 
      scroll_repeat_time = -1
      topFadeHidden = false
      gameinterface = 100012 -  (valid:true)
      updating_widgets = table: 12495E08
      lastx = 641
      total_fade_time = 1
      lasty = 401
      autosave_enabled = true
      OnMouseButton = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:339
      delayovercb = false
      blackoverlay = Image - images/global.xml:square.tex
      screenstack = table: 12493B30
      num_pending_saves = 0
      fade_delay_time = false
      fade_time = 1.0038603658477
      saving_indicator = UIAnim
      save_indicator_fade_time = 0
      alpha = -2.3011524197258e-007
      tracking_mouse = true
      focus_locked = false
      fade_title_in = false
      screenroot = screenroot
      consoletext = Text - CONSOLE TEXT
      repeat_time = 0.049995800107718
      displayingerror = false
      helptexttext = Text - 
   screen = ScriptErrorScreen
   forceOverBugScreen = nil
@E:/games/stream/steamapps/common/dont_starve/data/scripts/frontend.lua:747 in (method) ShowScreen (Lua) <744-749>
   self =
      topblackoverlay = Image - images/global.xml:square.tex
      fade_title_out = false
      ignoreups = table: 124945F8
      overlayroot = overlayroot
      save_indicator_time_left = 0
      fade_title_time = 0
      page_repeat_time = 0.25
      OnMouseMove = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:350
      helptextbg = Image - images/global.xml:square.tex
      helptext = HelpText
      subtitle = Text - 
      updating_widgets_alt = table: 156FFC20
      title = Text - 
      scroll_repeat_time = -1
      topFadeHidden = false
      gameinterface = 100012 -  (valid:true)
      updating_widgets = table: 12495E08
      lastx = 641
      total_fade_time = 1
      lasty = 401
      autosave_enabled = true
      OnMouseButton = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:339
      delayovercb = false
      blackoverlay = Image - images/global.xml:square.tex
      screenstack = table: 12493B30
      num_pending_saves = 0
      fade_delay_time = false
      fade_time = 1.0038603658477
      saving_indicator = UIAnim
      save_indicator_fade_time = 0
      alpha = -2.3011524197258e-007
      tracking_mouse = true
      focus_locked = false
      fade_title_in = false
      screenroot = screenroot
      consoletext = Text - CONSOLE TEXT
      repeat_time = 0.049995800107718
      displayingerror = false
      helptexttext = Text - 
   screen = ScriptErrorScreen
@E:/games/stream/steamapps/common/dont_starve/data/scripts/frontend.lua:775 in (method) DisplayError (Lua) <771-784>
   self =
      topblackoverlay = Image - images/global.xml:square.tex
      fade_title_out = false
      ignoreups = table: 124945F8
      overlayroot = overlayroot
      save_indicator_time_left = 0
      fade_title_time = 0
      page_repeat_time = 0.25
      OnMouseMove = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:350
      helptextbg = Image - images/global.xml:square.tex
      helptext = HelpText
      subtitle = Text - 
      updating_widgets_alt = table: 156FFC20
      title = Text - 
      scroll_repeat_time = -1
      topFadeHidden = false
      gameinterface = 100012 -  (valid:true)
      updating_widgets = table: 12495E08
      lastx = 641
      total_fade_time = 1
      lasty = 401
      autosave_enabled = true
      OnMouseButton = function - @E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2311942869/scripts/components/actionqueuer.lua:339
      delayovercb = false
      blackoverlay = Image - images/global.xml:square.tex
      screenstack = table: 12493B30
      num_pending_saves = 0
      fade_delay_time = false
      fade_time = 1.0038603658477
      saving_indicator = UIAnim
      save_indicator_fade_time = 0
      alpha = -2.3011524197258e-007
      tracking_mouse = true
      focus_locked = false
      fade_title_in = false
      screenroot = screenroot
      consoletext = Text - CONSOLE TEXT
      repeat_time = 0.049995800107718
      displayingerror = false
      helptexttext = Text - 
   screen = ScriptErrorScreen
@E:/games/stream/steamapps/common/dont_starve/data/scripts/mainfunctions.lua:971 in () ? (Lua) <948-1005>
   error = ...s/workshop-2081254154/scripts/descriptors/hunter.lua:23: attempt to index global 'TheWorld' (a nil value)
LUA ERROR stack traceback:
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:23 in (upvalue) GetAlternateBeastChance (Lua) <22-26>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:49 in (upvalue) GetHuntDataFromTrack (Lua) <38-51>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/descriptors/hunter.lua:67 in (field) Describe (Lua) <53-87>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/modmain.lua:813 in (upvalue) GetEntityInformation (Lua) <778-913>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/modmain.lua:952 in (upvalue) RequestEntityInformation (Lua) <916-1006>
    E:/games/stream/steamapps/common/dont_starve/data/../mods/workshop-2081254154/scripts/uioverrides.lua: [**truncated**]
   modnames = table: 1C2567A0
   modnamesstr = "  Simple Health Bar" "Combined Status" "[T] FastWork 2019" "99 Stack Items" "[T] Map Plugins(EN)" "Backpack and Amulet Slots" "Wormhole Marks" "Fast Travel" "Display Attack Range" "Zoom+ (Hamlet Compatible)" "egg_pain_queuer" " Storeroom" "Geometric Placement" "Insight" 
```

**Comment by penguin0616** [comment]
Fixed in 3a1d863d1509997abd52203b0b6658e718f35299, thanks for your report!

**Comment by yue-git** [comment]
It works，thanks
谢谢
